src/pip/_internal/utils/misc.py

```diff
@@ -1184,11 +1184,11 @@ def protect_pip_from_modification_on_windows(modifying_pip):
     On Windows, any operation modifying pip should be run as:
         python -m pip ...
     """
-    pip_names = [
-        "pip.exe",
-        "pip{}.exe".format(sys.version_info[0]),
-        "pip{}.{}.exe".format(*sys.version_info[:2])
-    ]
+    pip_names = set()
+    for ext in ('', '.exe'):
+        pip_names.add('pip{0}'.format(ext))
```

Comment:
Consistently use `{ext}`.

```suggestion
        pip_names.add('pip{ext}'.format(ext=ext))
```